azurerm/internal/services/storage/storage_account_resource.go

```diff
@@ -1563,7 +1564,21 @@ func resourceStorageAccountRead(d *pluginsdk.ResourceData, meta interface{}) err
 		d.Set("enable_https_traffic_only", props.EnableHTTPSTrafficOnly)
 		d.Set("is_hns_enabled", props.IsHnsEnabled)
 		d.Set("nfsv3_enabled", props.EnableNfsV3)
-		d.Set("allow_blob_public_access", props.AllowBlobPublicAccess)
+
+		if features.ThreePointOh() {
+			// There is a certain edge case that could result in the Azure API returning a null value for AllowBLobPublicAccess.
+			// Since the field is a pointer, this gets marshalled to a nil value instead of a boolean. Here we set this
+			// to the default value, but since this is a breaking change we'll enforce this in 3.0
+
+			allowBlobPublicAccess := getDefaultAllowBlobPublicAccess()
+			if props.AllowBlobPublicAccess != nil {
+				allowBlobPublicAccess = *props.AllowBlobPublicAccess
+			}
+			d.Set("allow_blob_public_access", allowBlobPublicAccess)
```

Comment:
I still think we need to rename this field in 3.0 to something more descriptive because it's confusing and doesn't accurately describe what it does (which is that nested items within this Account [e.g. Containers Blobs etc] can opt into being public) - maybe `can_nested_items_be_public` or something? The benefit of doing that is the Deprecated argument here would allow us to explain what this field _actually_ does?

Reply:
maybe "public_nested_items_enabled"

Reply:
I think that suffers from the same issue, I think we'll need to be a little wordy to be clear here tbh, given how this is misleading at the moment?

Reply:
what about `allow_public_nested_items` ?

Reply:
After further discussion with @tombuildsstuff and @redeux we decided to go with `allow_nested_items_to_be_public`.